### ISSUE TITLE
Date/time format accepted by templates

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/cicero#63]
The current implementation parses dates in the format `MM-DD-YYYY` which isn't ISO 8601 compliant. This may lead to issues when calling the moment library directly on those dates. Should Cicero allow for ISO 8601 format or automatically convert to it if required?

The corresponding part of the template processing is:
```
DATE -> MONTH "/" DAY "/" YEAR
{% (d) => {return '' + d[0] + '/' + d[2] + '/' + d[4]}%}
```

in `packages/cicero-core/lib/grammarvisitor.js`

**Comment by dselman** [comment]
We also need to make this locale aware, so that en-US templates use different date formatting than en-GB templates.

**Comment by dselman** [comment]
The parser generator should use a suitable date/time representation for the locale of the template. 

In addition the template author should be able to override the default date/time representation inline in the template grammar using a new `as` qualifier.

```
THIS CERTIFIES THAT in exchange for the payment by the undersigned purchaser (the “Purchaser”) of [{purchaseAmount}] (the “Purchase Amount”) on or about [{date as "YYYY-MM-DD HH:mm Z" }], [{company}], a [{state}] corporation (the “Company”), hereby issues to the Purchaser the right (the “Right”) to certain units of [{token}] (the “Token”), subject to the terms set forth below.
```

The date/time format string should correspond to a sensible (subset?) of the formatting strings supported by moment.js:
http://momentjs.com/docs/#/parsing/string-format/

**Comment by jphillips-bp3** [comment]
@dselman the `date as` approach would be perfect. In my particular case, the (US) client I am working with wants the format to be `MMMM d, YYYY`, for example `March 5, 2019`.

**Comment by hm2space** [comment]
In the same spirit, the localization should be applied to numbers too. 
For example 12,000 for 12000 doesn't pass the parsing.

**Comment by hm2space** [comment]
On the `date as`  it could be interesting to have some template level directive to set the date or num format.  And the 'as' at the field level would allow field level setting.

**Comment by dselman** [comment]
> In the same spirit, the localization should be applied to numbers too.
> For example 12,000 for 12000 doesn't pass the parsing.

Yes, the idea is that for numeric fields we could use the `as` to specify formatting.

**Comment by dselman** [comment]
> On the `date as` it could be interesting to have some template level directive to set the date or num format. And the 'as' at the field level would allow field level setting.

Yes, the default parsing rules will be set by the locale of the template (once implemented by #174)

**Comment by jeromesimeon** [comment]
Addressed in Cicero 0.12
